File: CONCEPTS/EXTRA/example_data_subset.py
import pandas as pd

# Define the DataFrame using vectors of literal data
df = pd.DataFrame([
    {'GAME_ID': 1000, 'TEAM_ID': 100, 'PLAYER_ID': 1, 'FGM': 10, 'FG3M': 12, 'FTM': 12},
    {'GAME_ID': 1000, 'TEAM_ID': 100, 'PLAYER_ID': 2, 'FGM':  4, 'FG3M':  4, 'FTM':  7},
    {'GAME_ID': 1000, 'TEAM_ID': 200, 'PLAYER_ID': 3, 'FGM':  2, 'FG3M':  6, 'FTM':  5},
    {'GAME_ID': 1000, 'TEAM_ID': 200, 'PLAYER_ID': 4, 'FGM':  8, 'FG3M':  2, 'FTM':  7},
    {'GAME_ID': 2000, 'TEAM_ID': 100, 'PLAYER_ID': 1, 'FGM': 10, 'FG3M':  4, 'FTM': 10},
    {'GAME_ID': 2000, 'TEAM_ID': 100, 'PLAYER_ID': 2, 'FGM': 11, 'FG3M':  5, 'FTM':  4},
    {'GAME_ID': 2000, 'TEAM_ID': 300, 'PLAYER_ID': 5, 'FGM':  8, 'FG3M': 10, 'FTM':  9},
    {'GAME_ID': 2000, 'TEAM_ID': 300, 'PLAYER_ID': 6, 'FGM':  7, 'FG3M':  6, 'FTM':  3},
])

print(df)

File: CONCEPTS/EXTRA/example_data_subset.R
# Define dataframe using literal vectors

df <- tibble::tribble(
  ~GAME_ID, ~TEAM_ID, ~PLAYER_ID, ~FGM, ~FG3M, ~FTM,
      1000,      100,          1,   10,    12,   12,
      1000,      100,          2,    4,     4,    7,
      1000,      200,          3,    2,     6,    5,
      1000,      200,          4,    8,     2,    7,
      2000,      100,          1,   10,     4,   10,
      2000,      100,          2,   11,     5,    4,
      2000,      300,          5,    8,    10,    9,
      2000,      300,          6,    7,     6,    3
)


# Display the dataframe
print(df)

# Different subset: boxscores has TEAM_ID = 400 which is not found in PLAYERs table
# Different subset: PLAYER has extra player not found in PLAYERs table

# Add a player with no boxscore data
players_df <- data.frame(
  PLAYER_ID = c(1, 2, 3, 4, 5, 6, 7),  # Added player 7
  PLAYER_NAME = c("Fred", "John", "Trevor", "Alex", "Jim", "Steve", "Michael")
)

# Add a boxscore with no player data
boxscores_df <- rbind(boxscores_df, 
                      data.frame(GAME_ID = 3000, TEAM_ID = 400, PLAYER_ID = 99, 
                                 FGM = 15, FG3M = 3, FTM = 8))